include/misc/__assert.h

```diff
@@ -86,10 +86,9 @@
 extern void posix_exit(int exit_code);
 #define __ASSERT_POST posix_exit(1)
 #else
+extern void k_panic(void);
```

Comment:
This is definitely not going to work, the build is going to look for a symbol at link time, but instead we need the expansion:

`#define k_panic()	_k_except_reason(_NANO_ERR_KERNEL_PANIC)`

Which gets expanded again as _k_except_reason is itself a macro.

Need to shuffle includes around to make this work.

Reply:
yes, it does not work. not sure why it compiled in one example i've tried initially. Instead of shuffling includes i could make assert.c file with `void assert_post_action()`. What do you think?

Reply:
I'm not quite sure I understand what you are proposing but the test you want to run to prove your changes work is tests/kernel/fatal